src/Paket.Core/RemoteUpload.fs

```diff
@@ -49,17 +91,24 @@ let GetUrlWithEndpoint (url: string option) (endPoint: string option) =
 
   
 let Push maxTrials url apiKey packageFileName =
-    let rec push trial =
+    let rec push trial = async {
         tracefn "Pushing package %s to %s - trial %d" packageFileName url trial
         try
             let client = Utils.createWebClient(url, None)
             client.Headers.Add("X-NuGet-ApiKey", apiKey)
-            client.UploadFileAsMultipart (new Uri(url)) packageFileName
-            |> ignore
+            let uploadAsync, progressReport = client.UploadFileAsMultipartAsync (new Uri(url)) packageFileName
+            use progressSubscription = 
+                progressReport 
+                |> Observable.sample (TimeSpan.FromSeconds(1.))
```

Comment:
instead of updating every second. How about updating when things changed? Like every 10%?